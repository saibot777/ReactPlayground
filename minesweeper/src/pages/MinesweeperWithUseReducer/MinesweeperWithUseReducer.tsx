import React, { FC } from 'react';

import { Top } from '@/components/Top';
import { GameLayout } from '@/components/Game';

import { GameWithUseReducer } from '@/modules/GameWithRedux';

export const MinesweeperWithUseReducer: FC = () => {
  return (
    <GameLayout
      top={
        <Top feature="Flag" firstAction="right click">
          Minesweeper with React+Redux and useReducer special for you
        </Top>
      }
    >
      <GameWithUseReducer />
    </GameLayout>
  );
};
