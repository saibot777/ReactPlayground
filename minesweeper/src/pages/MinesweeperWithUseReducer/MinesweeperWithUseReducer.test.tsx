import React from 'react';
import { render } from '@testing-library/react';

import { MinesweeperWithUseReducer } from './MinesweeperWithUseReducer';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
  useSearchParams: jest.fn().mockReturnValue([
    {
      get: () => null,
    },
    jest.fn(),
  ]),
}));

it('MinesweeperWithUseReducer renders correctly', () => {
  const { asFragment } = render(<MinesweeperWithUseReducer />);

  expect(asFragment()).toMatchSnapshot();
});
