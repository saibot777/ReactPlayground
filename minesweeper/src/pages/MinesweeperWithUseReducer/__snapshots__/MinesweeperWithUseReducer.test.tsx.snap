// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinesweeperWithUseReducer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  text-align: center;
  position: relative;
  display: inline-block;
}

.emotion-2 {
  font-size: 2em;
}

.emotion-3 {
  font-size: 1em;
  margin: 0 auto 2vw;
  line-height: 1.25em;
}

.emotion-4 {
  background: #e3e3e3;
}

.emotion-5 {
  color: #ec433c;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 6px solid #e3e3e3;
  background-color: #e3e3e3;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-8 {
  display: inline-block;
  padding: 0 0.3vw;
  color: #ec433c;
  border: 0.15vw inset;
  line-height: 2vw;
  letter-spacing: 0.08em;
  background: #333;
  text-shadow: 0 0 0.1vw #ec433c;
}

.emotion-9 {
  margin: 0;
  height: 100%;
  border-radius: 0;
  border: 0.15vw solid;
  border-color: white #9e9e9e #9e9e9e white;
  background-color: #d1d1d1;
}

.emotion-10 {
  font-weight: normal;
  display: block;
  white-space: nowrap;
  min-height: 1.2em;
  padding: 0 0.2vw 0.2vw;
}

.emotion-13 {
  font-size: 1.1vw;
  height: 100%;
  cursor: pointer;
  font-weight: 700;
  border-width: 0.15vw;
  border-style: solid;
  background-color: #d1d1d1;
  border-color: white #9e9e9e #9e9e9e white;
}

.emotion-15 {
  display: grid;
  grid-template-columns: repeat(9, auto);
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  padding: 1vw;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  width: 1.8vw;
  height: 1.8vw;
  color: transparent;
  background-color: #d1d1d1;
  border: 0.6vh solid transparent;
  border-color: white #9e9e9e #9e9e9e white;
}

.emotion-16:hover {
  -webkit-filter: brightness(1.1);
  filter: brightness(1.1);
}

<div
    class="emotion-0"
  >
    <header
      class="emotion-1"
    >
      <h1
        class="emotion-2"
      >
        Minesweeper with React+Redux and useReducer special for you
      </h1>
      <legend
        class="emotion-3"
      >
        <strong>
          Flag: 
        </strong>
        <code
          class="emotion-4"
        >
          <span
            class="emotion-5"
          >
            right click
          </span>
        </code>
      </legend>
    </header>
    <div
      class="emotion-6"
    >
      <div
        class="emotion-7"
      >
        <div
          class="emotion-8"
        >
          0
        </div>
        <div>
          <select
            class="emotion-9"
          >
            <option
              class="emotion-10"
              value="beginner"
            >
              beginner
            </option>
            <option
              class="emotion-10"
              value="intermediate"
            >
              intermediate
            </option>
            <option
              class="emotion-10"
              value="expert"
            >
              expert
            </option>
          </select>
          <button
            class="emotion-13"
          >
            ðŸ™‚
          </button>
        </div>
        <div
          class="emotion-8"
        >
          10
        </div>
      </div>
      <div
        class="emotion-15"
        role="grid"
        size="9"
      >
        <div
          class="emotion-16"
          data-testid="0,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="0,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="1,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="2,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="3,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="4,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="5,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="6,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="7,8"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,0"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,1"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,2"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,3"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,4"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,5"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,6"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,7"
          role="cell"
        >
          10
        </div>
        <div
          class="emotion-16"
          data-testid="8,8"
          role="cell"
        >
          10
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
